File: python/problem4.py
#!/usr/bin/python
n1 = 999
n2 = 999
res = []
for i in range(999, 1, -1):
	for j in range(1, 1000): 
		r = n1 * n2
		if str(r) == str(r)[::-1]:
			if r not in res:
				res.append(r)
		n1 -= 1
	n2 -= 1
	n1 = 999
res.sort()
print res[-1]

File: js/problem4.js
function strReverse(str) {
  return str.split("").reverse().join("");
}

function sortNum(a, b) {
  return a - b;
}

var n1 = 999, n2 = 999;
var res = [];

for(var i = 999; i > 1; i--){
  for(var j = 1; j < 1000; j++){
    var r = n1 * n2;
    if(r.toString() === strReverse(r.toString())) {
      if(!(r in res)){ 
		res.push(r);
      }
    }
	n1--;
  }
  n2--;
  n1 = 999;
}
res.sort(sortNum);
console.log(res[res.length - 1]);
